src/utils/wallet.js

```diff
@@ -1321,27 +1344,33 @@ const wallet = {
         }
       })
     } else {
+      // When it gets here, it means that already loaded all transactions
+      // so no need to load more
       if (resolve) {
         resolve();
       }
     }
 
     this.saveAddressHistory(historyTransactions, allTokens);
 
-    return {historyTransactions, allTokens, newSharedAddress, newSharedIndex};
+    return {historyTransactions, allTokens, newSharedAddress, newSharedIndex, addressesFound: lastGeneratedIndex + 1};
   },
 
   /**
    * Check if address is from the loaded wallet
    *
    * @param {string} address Address to check
+   * @param {Object} data Wallet data in localStorage already loaded
```

Comment:
I think you should explain that it is optional and should be used for performance improvements. If nothing is passed, the data will be loaded again.
We should also make clear that it is expected to receive the output of the `wallet.getWalletData()`.

Reply:
Added!